ophyd/areadetector/plugins.py

```diff
@@ -106,6 +112,24 @@ def array_pixels(self):
 
         return pixels
 
+    def read_configuration(self):
+        ret = super().read_configuration()
+
+        source_port = self.nd_array_port.get()
+        source_plugin = self.root.get_plugin_by_asyn_port(source_port)
```

Comment:
`source_plugin` seems like something that'd be handy to have as a property (or a `get_*` as you seem to prefer)